src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -84,38 +84,58 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
 }, {});
 
 const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelect }) => {
-  const [isNewSelectedPath, setIsNewSelectedPath] = useState(false);
-  const [previousSelectedPath, setPreviousSelectedPath] = useState(selectedPath);
   const [openKeys, setOpenKeys] = useState(openKeysToItem(menuItems[0], selectedPath));
+  const [currentNodeId, setCurrentNodeId] = useState();
+  const cursor = useRef(0);
   const selectedItem = useRef();
   const theme = useContext(ThemeContext);
+  const visibleNodes = [];
 
-  if (previousSelectedPath !== selectedPath) {
-    setIsNewSelectedPath(true);
-    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
-    setPreviousSelectedPath(selectedPath);
-  }
-
+  /**
+   * Scrolls the currently selected menu item into view on mount.
+   */
   useEffect(() => {
     if (selectedItem && selectedItem.current) {
       selectedItem.current.scrollIntoView();
     }
   }, []);
 
-  useLayoutEffect(() => {
-    if (!isNewSelectedPath) {
-      return;
-    }
-    const currentItemPosition = selectedItem?.current ? selectedItem.current.getBoundingClientRect() : null;
+  /**
+   * Updates cursor position when new menu items are loaded.
+   * If the item emphasized through keyboard navigation is not visible, scroll it into view.
+   */
+  useEffect(() => {
+    cursor.current = visibleNodes.findIndex((el) => el.id === currentNodeId);
+    const currentNode = currentNodeId ? document.querySelector(`#${currentNodeId}`) : null;
+    const currentNodePosition = currentNode ? currentNode.getBoundingClientRect() : null;
     const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
-    if (!currentItemPosition || !navigationMenuPosition) {
-      return;
+
+    if (currentNode) {
+      currentNode.focus();
+    }
+    if (currentNode && currentNodePosition && (currentNodePosition.bottom > navigationMenuPosition.bottom || currentNodePosition.top < navigationMenuPosition.top)) {
+      currentNode.scrollIntoView();
```

Comment:
One, or each, of these `scrollIntoView` calls is causing some unexpected page jumping. It's a little difficult to explain the steps but here it goes.

1. Navigate to the Secondary Nav Test Tab
2. Expand the Secondary Scroll Test node

![image](https://user-images.githubusercontent.com/6720991/86307484-7c402800-bbdc-11ea-9bf9-5e4aee60b673.png)

3. Use the arrow keys to highlight the Secondary Nav Test node

![image](https://user-images.githubusercontent.com/6720991/86307542-9f6ad780-bbdc-11ea-84b2-ca63d49d20aa.png)

4. Scroll to the bottom and mouse click the last nav item

Observe the page jumps back to the top. There's different combinations of arrow keys and mouse clicks the navigation menu jumps around

Reply:
My latest commit should resolve this issue.

Reply:
I'm unclear, what is the purpose of the scrollIntoView? Setting focus does this automatically, so is this needed for an expand/collapse all behavior, or if this for jumping immediately to a specific page?

Reply:
I didn't know that `focus()` automatically scrolled into view and I just never tried it without the `scrollIntoView` call. I've just refactored to remove the extraneous calls.